pkg/storage/ingest/fetcher.go

```diff
@@ -48,6 +48,9 @@ type fetcher interface {
 
 	// Stop stops the fetcher.
 	Stop()
+
+	// BufferedRecords returns the number of records that have been fetched but not yet consumed.
+	BufferedRecords() float64
```

Comment:
[nit]Design wise, this should return `int64` given the number of buffered records is a discrete number. Then we convert to `float64` in the gauge func.

Reply:
Done in https://github.com/grafana/mimir/pull/9855/commits/43f6386c4e650ca5b22d651d4ba0a7cb60d9672d